deploy/olm-catalog/service-telemetry-operator/metadata/properties.yaml

```diff
@@ -1,3 +1,44 @@
 properties:
   - type: olm.maxOpenShiftVersion
     value: "4.12"
+  - type: olm.constraint
+    value:
+      failureMessage: Require Smart Gateway for Service Telemetry Framework
+      all:
+        constraints:
+          - failureMessage: Package smart-gateway-operator is needed for Service Telemetry Framework
+            package:
+              packageName: smart-gateway-operator
+              versionRange: '>=5.0.0'
+  - type: olm.constraint
+    value:
+      failureMessage: Require data transport for Service Telemetry Framework
+      all:
+        constraints:
+          - failureMessage: Package amq7-interconnect-operator is needed for data transport with STF
+            package:
+              packageName: amq7-interconnect-operator
+              versionRange: '>=1.10.0'
+  - type: olm.constraint
+    value:
+      failureMessage: Require certificate management for Service Telemetry Framework
+      all:
+        constraints:
+          - failureMessage: Package openshift-cert-manager-operator is needed for AMQ Interconnect setup
+            package:
+              packageName: openshift-cert-manager-operator
+              versionRange: '>=1.10.0'
+  - type: olm.constraint
+    value:
+      failureMessage: Require Prometheus backend for data storage of metrics for Service Telemetry Framework
+      any:
+        constraints:
+          - package:
+              packageName: prometheus
+              versionRange: '>= 0.56.0'
+          - package:
+              packageName: rhods-prometheus-operator
```

Comment:
I didn't test this in #421. I'd expect it to work, but to be treated as the "community" operator (since IIRC it provides the prometheus.monitoring.coreos.com API). So for example if OBO isn't available in the catalog, `observability_strategy: use_redhat`, is going to fail. I think this might be a good thing, all things considered, because if I understand correctly

Reply:
Yea, so I expect this Operator to just go away at some point, but if we're going to have access to a Prometheus Operator that provides the coreos.com API in the Red Hat CatalogSource, that does seem like my preference overall.

I could also be convinced to just drop this, and not have this as a dependency at all though and stick with what is available in Community Operators.

Reply:
I think I'm actually going to entirely remove rhods from here. I feel like it's going to cause us problems in the future. There are ongoing discussions about having it removed from the catalog. It is also not going to be supported outside the context of RHODS, so there isn't really any reason to set ourselves up for a future problem.